tests/observability/suite.go

```diff
@@ -271,3 +280,145 @@ func testPodDisruptionBudgets(check *checksdb.Check, env *provider.TestEnvironme
 
 	check.SetResult(compliantObjects, nonCompliantObjects)
 }
+
+// Function to convert APIRequestCount objects into ToBeRemovedWorkloadAPI
+// Filters:
+// - status.removedInRelease is not empty
+// - Verifies if the service account is inside the workload SA list from env.ServiceAccounts
+func buildToBeRemovedWorkloadAPIList(apiRequestCounts []apiserv1.APIRequestCount, workloadServiceAccountNames map[string]struct{}) []provider.ToBeRemovedWorkloadAPI {
+	var toBeRemovedWorkloadAPIs []provider.ToBeRemovedWorkloadAPI
+
+	for i := range apiRequestCounts {
+		obj := &apiRequestCounts[i]
+		// Filter by non-empty removedInRelease
+		if obj.Status.RemovedInRelease != "" {
+			workloadAPI := provider.ToBeRemovedWorkloadAPI{
+				APIName:          obj.Name,
+				RemovedInRelease: obj.Status.RemovedInRelease,
+				UserInfo:         make(map[string]struct{}),
+			}
+
+			// Iterate over the last 24h usage data
+			for _, last24h := range obj.Status.Last24h {
+				for _, byNode := range last24h.ByNode {
+					for _, byUser := range byNode.ByUser {
+						// Split the username by ":" and take the last chunk
+						serviceAccountParts := strings.Split(byUser.UserName, ":")
+						strippedServiceAccount := serviceAccountParts[len(serviceAccountParts)-1]
+
+						// Check if the service account is in the workload SA list
+						if _, exists := workloadServiceAccountNames[strippedServiceAccount]; exists {
+							userInfo := fmt.Sprintf("UserName: %s, UserAgent: %s", byUser.UserName, byUser.UserAgent)
+							workloadAPI.UserInfo[userInfo] = struct{}{}
```

Comment:
Hi @greyerof I like your idea, just added the release version in which it will be removed as the key
```
// Define a map where the key is the service account name and the value is another map
// The inner map key is the API name and the value is the release version in which it will be removed
serviceAccountToDeprecatedAPIs := make(map[string]map[string]string)
```